src/main/java/com/onarandombox/MultiverseCore/commands/DumpsCommand.java

```diff
@@ -0,0 +1,320 @@
+
+        MVVersionEvent versionEvent = new MVVersionEvent();
+
+        this.addDebugInfoToEvent(versionEvent);
+        plugin.getServer().getPluginManager().callEvent(versionEvent);
+
+        // Add plugin list if user isn't paranoid
+        if (!parsedFlags.hasFlag("--paranoid")) {
+            versionEvent.putDetailedVersionInfo("plugins.md", "# Plugins\n\n" + getPluginList());
+        }
+
+
+        // Deal with default case of logs
+        LogsType logsType = parsedFlags.flagValue("--logs", LogsType.class);
+        if (logsType == null) {
+            logsType = LogsType.MCLOGS;
+            Logging.finer("Logs was null so set to mclogs");
+        }
+
+
+
+
+        // Deal with default case of upload
+        Services services = parsedFlags.flagValue("--upload", Services.class);
+        if (services == null) {
+            services = Services.PASTEGG;
+            Logging.finer("Upload was null so set to pastegg");
+        }
+
+
+        // Need to be final for some reason...
+        final LogsType finalLogsType = logsType;
+        final Services finalServices = services;
+        BukkitRunnable logPoster = new BukkitRunnable() {
+            @Override
+            public void run() {
+                HashMap<String, String> pasteURLs = new HashMap<>();
+                Logging.finer("Logs type is: " + finalLogsType);
+                Logging.finer("Services is: " + finalServices);
+
+                // Deal with logs flag
+                if (!parsedFlags.hasFlag("--paranoid")) {
+                    switch (finalLogsType) {
+                        case MCLOGS -> {
+                            issuer.sendInfo(MVCorei18n.DUMPS_UPLOADING_LOGS, "{link}", "https://mclo.gs");
+                            pasteURLs.put("Logs", postToService(PasteServiceType.MCLOGS, false, getLogs(), null));
+                        }
+                        case APPEND -> {
+                            versionEvent.putDetailedVersionInfo("latest.log", getLogs());
+                        }
+                    }
+                }
+
+                // Get the files from the event
+                final Map<String, String> files = versionEvent.getDetailedVersionInfo();
+
+                // Deal with uploading debug info
+                switch (finalServices) {
+                    case PASTEGG -> {
+                        issuer.sendInfo(MVCorei18n.DUMPS_UPLOADING, "{link}", "https://paste.gg");
+                        pasteURLs.put("paste.gg", postToService(PasteServiceType.PASTEGG, true, null, files));
+                    }
+                    case PASTESDEV -> {
+                        issuer.sendInfo(MVCorei18n.DUMPS_UPLOADING, "{link}", "https://pastes.dev");
+                        pasteURLs.put("pastes.dev", postToService(PasteServiceType.PASTESDEV, true, null, files));
+                    }
+                }
+
+                // Finally, loop through and print all URLs
+                for (String service : pasteURLs.keySet()) {
```

Comment:
I dont really see a need for loop, bcu its just printing 1 or 2 things

Reply:
Good point. I might just make it print out directly